### ISSUE TITLE
Cache not being utilized when called through Docker build

### ISSUE THREAD

**Issue by mkotsollaris** [fox1t/turborepo-remote-cache#34]
I've successfully cloned the `turborepo-remote-cache` and I've hooked it up in AWS. Everything is working fine, up until I try to use the cache from docker. Here's my `dockerfile`:

```jsx
FROM docker.artifactory.moveaws.com/node:16 as build
COPY . /app
WORKDIR /app
RUN yarn install
RUN yarn build
RUN npx turbo run build
```

Both `RUN yarn build` and `RUN npx turbo run build` are not updating the cache. Originally I thought that my docker instance couldn't communicate with the deployed service, but when I added a CURL inside the container, I could verify that outbound calls are properly triggered.

My question is, has anyone run into this? Maybe someone has an example of a `dockerfile` already out there? Or what could possibly be the issue that the commands won't trigger the cache during the build phase?

**Comment by fox1t** [comment]
You can follow here https://github.com/fox1t/turborepo-remote-cache#deploy-on-docker
You are not setting the vars. 

You can also use the official image we pushed on docker hub.

**Comment by fox1t** [comment]
You can find the Dockerfile here: https://github.com/fox1t/turborepo-remote-cache/blob/main/Dockerfile

**Comment by StevenMatchett** [comment]
@fox1t I believe @mkotsollaris is referring to building a docker image that uses the remote-cache and not deploying the remote cache. I am also running into issues when trying to reach out to my deployed remote-cache. When docker is running a turbo command it is not using hosted remote cache

**Comment by fox1t** [comment]
every S3 bucket is a remote cache, no matter where the docker image runs. I am not sure why you can’t use the provided docker image. What are the limitations?

**Comment by mkotsollaris** [comment]
> Ok, I finally got it. You are trying to connect to a remotely deployed remote-cache server from a docker container. Right?

Yes, we are basically just testing this implementation. It works great up until we try use remote-caching during `docker build`, for some reason it won't work! 🤔

**Comment by fox1t** [comment]
Can you check if the same setup works with Vercel's own remote caching? We should exclude some kind of unexpected setting/bug on the turborepo client-side.

**Comment by fox1t** [comment]
ref: https://github.com/vercel/turborepo/issues/1508

**Comment by fox1t** [comment]
@StevenMatchett, I couldn't make it work with your hack. Can you please write every step you took to make it work?

**Comment by fox1t** [comment]
I've just done some experiments, and it turned out that you don't need VERCEL_ vars (for me, turborepo-remote-cache is not working using VERCEL_ vars).
I added my findings to the README: https://github.com/fox1t/turborepo-remote-cache#enable-remote-caching-in-docker
Also, we needed to update turborepo-remote-cache to version 1.4, introducing the support for the `slug` query string parameter. Unfortunately, when you use the `--team` CLI option, it passes it as `slug` instead of `teamId`. I don't know why they decided like that or if it is just a bug. This is the ref: https://github.com/vercel/turborepo/blob/d83130a8b30b01ae2ce1627f4188e791598ea8b3/cli/internal/config/config.go#L172

**Comment by StevenMatchett** [comment]
Awesome thank you! I am curious if its a linux thing rather than a docker thing

**Comment by StevenMatchett** [comment]
@fox1t here is a PR for the fix in turborepo

https://github.com/vercel/turborepo/pull/1527

**Comment by LucM** [comment]
FIY - I still had the problem with the last updates.

With the `node:16-slim` turbo was not fetching cache from the remote server. 

-> I had to switch for a `node:16-alpine`.

I do not know the reason, however it could be interesting to add it in the documentation :)

**Comment by NorkzYT** [comment]
@LucM 

What Architecture are you running your setup on?

**Comment by rahnarsson** [comment]
Hello! 

Had this similar issue myself and after some Wiresharking the connection between my local machine and turbo-cache I noticed that the TLS Handshake failed between my docker build and remote cache. No error was produced on the build logs about bad certificate. The certificate I use on remote cache is valid AWS Issued certificate. 

I used node:18-slim base image and after installing ca-certificates package:

```
RUN apt-get update && apt-get install -y \
    ca-certificates \
&& rm -rf /var/lib/apt/lists/*
```

I was able to hit and connect to cache.
